contracts/BridgeCommittee.sol

```diff
@@ -128,20 +155,23 @@ contract BridgeCommittee {
         view
         returns (bool)
     {
-        // Get the message type from the message hash.
+        // I don't quite like we decode the same message a few times to get different fields. Can 
```

Comment:
good point. it felt more readable but probably not the most efficient approach. will go back to the drawing board on this.